profile/src/main/java/com/scottlogic/deg/profile/reader/atomic/ConstraintValueValidator.java

```diff
@@ -0,0 +1,188 @@
+
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+import java.util.regex.Pattern;
+
+import static com.scottlogic.deg.profile.dto.AtomicConstraintType.IS_NULL;
+
+public class ConstraintValueValidator {
+
+    public static void validate(String field, AtomicConstraintType type, Object value){
+        try {
+            validateConstraintValue(type, value);
+        } catch (IllegalArgumentException | ValidationException e){
+            throw new InvalidProfileException(String.format("Field [%s]: %s", field, e.getMessage()));
+        }
+    }
+
+    private static void validateConstraintValue(AtomicConstraintType type, Object value){
+
+        if (type == IS_NULL){
+            validateNull(value);
+            return;
+        }
+
+        validateNotNull(value);
+
+        switch (type) {
+            case IS_EQUAL_TO_CONSTANT:
+                validateAny(type, value);
+                break;
+            case IS_IN_SET:
+                validateSet(type, value);
+                break;
+            case IS_OF_TYPE:
+                validateTypes(value);
+                break;
+
+            case MATCHES_REGEX:
+            case CONTAINS_REGEX:
+                validatePattern(value);
+                break;
+
+            case HAS_LENGTH:
+            case IS_STRING_SHORTER_THAN:
+            case IS_STRING_LONGER_THAN:
+                validateStringLengthInt(type, value);
+                break;
+
+            case IS_GREATER_THAN_CONSTANT:
+            case IS_GREATER_THAN_OR_EQUAL_TO_CONSTANT:
+            case IS_LESS_THAN_CONSTANT:
+            case IS_LESS_THAN_OR_EQUAL_TO_CONSTANT:
+                validateNumber(type, value);
+                break;
+
+            case IS_AFTER_CONSTANT_DATE_TIME:
+            case IS_AFTER_OR_EQUAL_TO_CONSTANT_DATE_TIME:
+            case IS_BEFORE_CONSTANT_DATE_TIME:
+            case IS_BEFORE_OR_EQUAL_TO_CONSTANT_DATE_TIME:
+                validateDateTime(value);
+                break;
+
+            case IS_GRANULAR_TO:
+                validateGranularity(value);
+        }
+    }
+
+    private static void validateNull(Object value) {
+    }
+
+    private static void validateNotNull(Object value) {
+        if (value == null){
+            throw new ValidationException("Couldn't recognise 'value' property, it must be set to a value");
+        }
+    }
+
+    private static void validateAny(AtomicConstraintType type, Object value) {
+        if (value instanceof OffsetDateTime) {
+            validateDateTime(value);
+        }
+        else if (value instanceof Number){
+            validateNumber(type, value);
+        }
+    }
+
+    private static void validateSet(AtomicConstraintType type, Object value) {
+        if (!(value instanceof DistributedSet)){
+            throw new ValidationException("Couldn't recognise 'values' property, it must not contain 'null'");
+        }
+
+        if (((DistributedSet) value).set().isEmpty()) {
```

Comment:
not sure i like us always going through .set() in DistributedSet.  In fact i'd rather we didn't go through it at all unless we really need the full collected Set.  I'd quite like us to add isEmpty and stream() to the interface and have the concrete classes implement them sensibly, avoiding collecting all values into a Set<T> first